org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -364,6 +370,22 @@ class CGenerator extends GeneratorBase {
         println('******** generated binaries: ' + fileConfig.binPath)
     }
 
+    /**
+     * 
+     */
+    override setTargetConfig(IGeneratorContext context) {
+        super.setTargetConfig(context);
+        // Set defaults for the compiler after parsing the target properties
+        // of the main .lf file.
+        if(this.CCppMode && targetConfig.useCmake == false && targetConfig.compiler.isNullOrEmpty) {
```

Comment:
Better not to repeat the same conditional twice. Cleaner to have something like:
```
if (targetConfig.useCmake == false && targetConfig.compiler.isNullOrEmpty) {
  if (this.CCppMode) {
    // ..
  } else {
    // .. 
  }
}
```

Reply:
Thanks, that's a great suggestion.